app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt

```diff
@@ -588,6 +578,36 @@ class ComposeViewModel @Inject constructor(
         CONTINUE_EDITING_OR_DISCARD_CHANGES, // editing post
         CONTINUE_EDITING_OR_DISCARD_DRAFT // edit draft
     }
+
+    data class QueuedMedia(
+        val localId: Int,
+        val uri: Uri,
+        val type: Type,
+        val mediaSize: Long,
+        val uploadPercent: Int = 0,
+        val id: String? = null,
+        val description: String? = null,
+        val focus: Attachment.Focus? = null,
+        val state: State
+    ) {
+        enum class Type {
+            IMAGE,
+            VIDEO,
+            AUDIO
+        }
+        enum class State {
+            UPLOADING,
+            UNPROCESSED,
+            PROCESSED,
+            PUBLISHED
+        }
+    }
+
+    data class PickedMedia(
```

Comment:
Naming: Is it important that it is "picked" here? Or more like "LocalMedia" or somesuch.